### DISCUSSION TITLE
What does 'Lgb' mean in 'LgbExtractor' and where is allLgbFiles

### DISCUSSION THREAD

**Discussion by elton11220** [ffxiv-teamcraft/ffxiv-teamcraft#2592] [Q&A] [answered]
LgbFolder does not appear to exist
[https://github.com/ffxiv-teamcraft/ffxiv-teamcraft/blob/d2c258939385682fb38f0af425cb7d9ddc7d0d9e/apps/data-extraction/src/extractors/lgb.extractor.ts#L77](https://github.com/ffxiv-teamcraft/ffxiv-teamcraft/blob/d2c258939385682fb38f0af425cb7d9ddc7d0d9e/apps/data-extraction/src/extractors/lgb.extractor.ts#L77)

**Comment by Supamiu** [comment]
Hi, what are you using this for? That folder is intended to be populated using an external tool exe that I can share.

And LGB stands for `Layer Group Binary`, it's used to fill a lot of data regarding NPC  and Fate positions.

EDIT: please feel free to join the Discord server and ping @Miu#1568 on it, I'm pretty available and it'd be much easier to answer and help on discord 😄 https://discord.gg/r6qxt6P

**Reply by elton11220** [reply]
Thanks for answering. I'm a React Native learner. Since I love this game so much, I developed a gathering clock app(Android) for practice. The process of the development is finished, and I plan to add a function to display the nearest Aetheryte to the gathering point. I have researched the tables "Aetheryte", "Map", "MapMarker", "TerritoryType", "ExportedGatheringPoint" and then I found coordinates that related to aetheryte might be in the "MapMarker" table. But I can't find any association of the coordinates in "ExportedGatheringPoint" and in the "MapMarker" table. So I started digging into the source code of teamcraft and eventually found out that the coordinates of the aetheryte might be parsed in lgbExtractor. In the process of reading the source code, I found that the coordinates of the aetheryet are parsed in the following code.[https://github.com/ffxiv-teamcraft/ffxiv-teamcraft/blob/staging/apps/data-extraction/src/extractors/lgb.extractor.ts#L76-L123](https://github.com/ffxiv-teamcraft/ffxiv-teamcraft/blob/staging/apps/data-extraction/src/extractors/lgb.extractor.ts#L76-L123) I want to learn how the coordinates are converted, but I was stuck in the process of finding the lgb file.
I would appreciate it if you tell me the solution or the way that I can get the lgb file

**Reply by Supamiu** [reply, marked as answer]
I really recommend you to use the nodes.json lazy file directly (https://github.com/ffxiv-teamcraft/ffxiv-teamcraft/blob/staging/libs/data/src/lib/json/nodes.json) and load it from github's [CDN](https://raw.githubusercontent.com/ffxiv-teamcraft/ffxiv-teamcraft/staging/libs/data/src/lib/json/nodes.json) (or downloading it and hosting it yourself or including it in the app).

You could the combine it with the Aetheryte lazy file (https://github.com/ffxiv-teamcraft/ffxiv-teamcraft/blob/staging/libs/data/src/lib/json/aetherytes.json) do achieve what you want.

These files are updated everytime the game client is updated.

If you really want to use the LGB files, you can grab them using https://github.com/lmcintyre/LGBToJson and then you have to copy them in the lgb folder.
